### PR TITLE
Create Java Solution for Milk Factory

### PR DESCRIPTION
**Pull request by MrinallU**

### CODE REVIEW

**Comment by MrinallU** [comment]
@bqi343 I think this should be good to merge.

File: solutions/USACO open19-milk-factory.mdx [hunk 1]
```diff
@@ -1,17 +1,103 @@
...
+    public static void main(String[] args) throws IOException {
+        Scanner sc = new Scanner(new File("factory.in"));
+        PrintWriter out = new PrintWriter(new BufferedWriter(new FileWriter("factory.out")));
+        int N = sc.nextInt(); Graph  g = new Graph(N); boolean used = false;
+        for (int i = 0; i < N - 1; i++) {
+            g.addEdge(sc.nextInt() - 1, sc.nextInt() - 1); // Meaning one can walk from station A to station B. 
+        }
+        for (int i = 0; i < N; i++) {
+            int ans = g.DFS(i);
+            if(ans >= N) {
```
Comment (frodakcin) [inline solutions/USACO open19-milk-factory.mdx:82]:
```suggestion
            if(ans == N) {
```

Comment (frodakcin) [reply solutions/USACO open19-milk-factory.mdx:82]: For clarity

File: solutions/USACO open19-milk-factory.mdx [hunk 2]
```diff
@@ -1,17 +1,103 @@
...
+    }
+
+    // The function to do DFS traversal. It uses recursive DFSUtil()
+    int DFS(int v)
+    {
+        // Mark all the vertices as not visited(set as false by default)
+        boolean visited[] = new boolean[V];
+
+        // Call the recursive helper function to print DFS traversal
+        return DFSUtil(v, visited);
```
Comment (frodakcin) [inline solutions/USACO open19-milk-factory.mdx:68]: This array is being passed by reference, right? (as opposed to copy)

**Review by frodakcin — CHANGES_REQUESTED** [review]
Mostly good. Just a slight change I'd like

**Comment by MrinallU** [comment]
@frodakcin Alright that is fixed! Thanks for pointing that one out.

File: solutions/USACO open19-milk-factory.mdx (hunk 2 above)
Comment (MrinallU) [reply solutions/USACO open19-milk-factory.mdx:68]: @frodakcin I believe it is.

**Review by frodakcin — APPROVED** [review]
Gonna wait for a couple more reviews

**Review by dongliu0426 — APPROVED** [review]

**Review by Farrius — APPROVED** [review]

File: solutions/USACO open19-milk-factory.mdx [hunk 3]
```diff
@@ -1,17 +1,103 @@
...
+        }
+        for (int i = 0; i < N; i++) {
+            int ans = g.DFS(i);
+            if(ans == N) {
+                out.println(i + 1); // If the number of visited stations from the DFS is equal to the total number of stations then output. 
+                used = true;
+                break;
+            }
+        }
+        if(!used){
```
Comment (bqi343) [inline solutions/USACO open19-milk-factory.mdx:88]: bugs me that you don't seem to be consistent about spacing D:

`for (int i = 0; i < N; i++) {`

`if(ans == N) {`

`if(!used){`

```
while (i.hasNext())
        {
```

Why is there so much space in the `addEdge` function?

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/USACO open19-milk-factory.mdx (hunk 3 above)
Comment (darren-yao) [reply solutions/USACO open19-milk-factory.mdx:88]: @MrinallU I'll approve once you fix formatting and spacing consistency.

**Comment by MrinallU** [comment]
@bqi343 The spacing should be consistent now. I use Vm so I haven't found a good formatting plugin yet, so I apologize for the formatting inconsistencies.

File: solutions/USACO open19-milk-factory.mdx [hunk 4]
```diff
@@ -1,17 +1,95 @@
...
+         }
+         for (int i = 0; i < N; i++){
+              int ans = g.DFS(i);
+              if(ans == N){
+                   out.println(i + 1); // If the number of visited stations from the DFS is equal to the total number of stations then output. 
+                   used = true;
+                   break;
+              }
+         }
+         if(!used){
```
Comment (bqi343) [inline solutions/USACO open19-milk-factory.mdx:82]: spacing for these if statements are different from the one above

**Review by bqi343 — CHANGES_REQUESTED** [review]

**Comment by MrinallU** [comment]
@bqi343 How about now?

File: solutions/USACO open19-milk-factory.mdx [hunk 5]
```diff
@@ -1,17 +1,95 @@
...
+import java.io.*;
+import java.util.*;
+
+class Graph{
+     private int V;
+     private ArrayList<ArrayList<Integer>> adj;
+     Graph(int V){
+         this.V = V;
+         adj = new ArrayList<ArrayList<Integer>>(V);
+         for (int i = 0; i < V; i++){
```
Comment (bqi343) [inline solutions/USACO open19-milk-factory.mdx:30]: now this for loop is different :(

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/USACO open19-milk-factory.mdx (hunk 5 above)
Comment (MrinallU) [reply solutions/USACO open19-milk-factory.mdx]: Oh, I should be good now

File: solutions/USACO open19-milk-factory.mdx [hunk 6]
```diff
@@ -1,17 +1,95 @@
...
+              adj.add(new ArrayList<Integer>()); // Creating the adjacency list
+         }
+     }
+     void addEdge(int u, int v){
+          adj.get(v).add(u); // Connecting the two factories.
+     }
+    
+
+     int DFSUtil(int v,boolean visited[]){
+         int ans  = 0;
```
Comment (dongliu0426) [inline solutions/USACO open19-milk-factory.mdx:40]:
```suggestion
         int ans = 0;
```

**Review by dongliu0426 — CHANGES_REQUESTED** [review]

**Review by darren-yao — APPROVED** [review]
ok

**Pull request merged**